### DISCUSSION TITLE
TP8 ejercicio 1 eliminar primer elemento

### DISCUSSION THREAD

**Discussion by luciano-repetti** [INGCOM-UNRN-P1/cursada-2023#202] [Preguntas y respuestas] [answered]
Buenas! Estoy intentando hacer la función: ```void eliminar_del_principio(nodo_t* lista);``` pero al hacerlo solo se me borra dentro de la funcion ya que "lista" es la copia.

Esta es mi función:

```
void eliminar_del_principio(nodo_t* lista)
{
    if(lista != NULL)
    {
        nodo_t *nodo_temp = lista->siguiente;
        lista = nodo_temp;
        imprimir_lista(lista);
    }
}
```

Dos formas se me ocurrieron pero implican modificar el prototype, por lo cuál no se si lo estoy haciendo mal.

PD: tengo imprimir_lista en la función para saber si ocurre lo que quiera hacer dentro del bloque.

**Comment by Nikzza** [comment]
https://github.com/orgs/INGCOM-UNRN-P1/discussions/200, tal vez te sirva

**Reply by luciano-repetti** [reply]
Si lo vi lo del doble puntero, tiene sentido, pero es modificando el prototype o si no de devolver la lista de nodos modificada. Igualmente se me acaba de ocurrir crearla con memoria dinamica, tal vez me funcione. Si tenes otra forma te agradecería cooperación

**Comment by Nikzza** [comment]
en mi caso modifique el prototype para que funcione. La de memoria dinamica tambien es buena, no se si funcione. La utilizacion del doble puntero esta justificada, lo que no se es si en la clase de hoy dio el indicio de si podiamos modificar el prototype pero dio un ejemplo de porque debiamos usar el doble puntero

**Reply by luciano-repetti** [reply, marked as answer]
Perfecto! Entonces modificamos el prototype un poco. Gracias
